runtime/compiler/p/codegen/PPCJNILinkage.cpp

```diff
@@ -433,8 +433,8 @@ TR::Register *J9::Power::JNILinkage::buildDirectDispatch(TR::Node *callNode)
 
       // push flags: use lis instead of lis/ori pair since this is a constant. Save one instr
       aValue = fej9->constJNICallOutFrameFlags();
-      TR_ASSERT((aValue & ~0x7FFF0000) == 0, "Length assumption broken.");
-      generateTrg1ImmInstruction(cg(), TR::InstOpCode::lis, callNode, gr11Reg, (aValue>>16)&0x0000FFFF);
+      TR_ASSERT_FATAL((aValue & ~0x7FFF0000) == 0, "Length assumption broken.");
+      generateTrg1ImmInstruction(cg(), TR::InstOpCode::lis, callNode, gr11Reg, aValue>>16);
```

Comment:
I am not sure the ASSERT above guaranties there is nothing in the upper 32-bits of aValue (intptr_t). I would cast to int16_t to be on a safe side and to reproduce the old behaviour precisely. Or make the ASSERT more clear.

Reply:
This assert does guarantee that the upper bits are unset, but thinking about it, I do agree that the type conversions involved can be a bit unintuitive. This works based on sign-extension to `intptr_t` after negation, so it only works because the upper bit of that mask is set after negating it. Perhaps using `(aValue & ~static_cast<intptr_t>(0x7FFF0000))` instead would make this more clear?

Reply:
Opened https://github.com/eclipse/openj9/issues/9930